<h1 style="text-align:center;"><b>Laboratorio 2</b></h1>
<h3 style="text-align:center;">Marcos Díaz (221102), Daniel Machic (22118), Maria Jose Ramírez (221051)</h3>

**Github**: https://github.com/mac2218/Seminario-Lab02

Librerias

In [2]:
import numpy as np

In [1]:


# Probabilidades conjuntas P(X, Y)
joint = np.array([
    [0.09, 0.16, 0.17, 0.05],  # Y=0
    [0.15, 0.07, 0.10, 0.21]   # Y=1
])


px = joint.sum(axis=0)


posterior = joint / px


decision = np.argmax(posterior, axis=0)

# Error de Bayes
error = sum(px[i] * np.min(posterior[:, i]) for i in range(len(px)))

print("Posteriores P(Y|X):\n", posterior)
print("\nRegla óptima (0=Y0, 1=Y1):\n", decision)
print("\nError de Bayes:", error)


Posteriores P(Y|X):
 [[0.375      0.69565217 0.62962963 0.19230769]
 [0.625      0.30434783 0.37037037 0.80769231]]

Regla óptima (0=Y0, 1=Y1):
 [1 0 0 1]

Error de Bayes: 0.31


## Clasificador Bayesiano Óptimo

### 1. Probabilidades posteriores

A partir de la tabla de probabilidades conjuntas \( P(X,Y) \), calculamos:
$P(Y \mid X=x) = \frac{P(X=x, Y)}{P(X=x)}$

Las probabilidades posteriores obtenidas son:

$
P(Y \mid X)$ =
\begin{bmatrix}
0.375 & 0.6957 & 0.6296 & 0.1923 \\
0.625 & 0.3043 & 0.3704 & 0.8077
\end{bmatrix}


donde la primera fila corresponde a \( Y=0 \) y la segunda a \( Y=1 \).

---

### 2. Regla de clasificación óptima

El clasificador Bayesiano asigna la clase con mayor probabilidad posterior:

$hat{y}(x) = \arg\max_{y \in \{0,1\}} P(Y=y \mid X=x)$

Por lo tanto:

$
\hat{y}(1)=1, \quad
\hat{y}(2)=0, \quad
\hat{y}(3)=0, \quad
\hat{y}(4)=1
$

---

### 3. Error de Bayes

El error mínimo de clasificación está dado por:

$
P_e = \sum_{x} P(X=x)\min \{P(Y=0 \mid X=x), P(Y=1 \mid X=x)\}
$
Sustituyendo valores:

$
P_e = 0.31
$

Por lo tanto, el error del clasificador Bayesiano óptimo es:

$
\boxed{31\%}
$
